clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -183,4 +197,50 @@ public synchronized long lastSuccessfulUpdate() {
     public long refreshBackoff() {
         return refreshBackoffMs;
     }
+
+    /**
+     * Set state to indicate if metadata for all topics in Kafka cluster is required or not.
+     * @param needMetadaForAllTopics boolean indicating need for metadata of all topics in cluster.
+     */
+    public void needMetadataForAllTopics(boolean needMetadaForAllTopics) {
+        this.needMetadataForAllTopics = needMetadaForAllTopics;
```

Comment:
Yeah, I guess that makes sense. It kind of feels weird to be tracking a topic list in Metadata that we're not going to use, however. My feeling is that maybe we shouldn't update Metadata for topics subscribed from a regex. Instead we just update metadata once in subscribe(pattern). Does that make sense?